src/js/ui/createPost/createPost.mjs

```
@@ -0,0 +1,39 @@
+import { sendPost } from "../../data/API/sendPost.mjs";
+import { updateOldObject } from "../notifications/updateOldObject.mjs";
+
+/**
+ * This function gathers inputs from form field inputs and sends a
+ * create post request to the API via the sendPost function.
+ * It clears the form fields after successfully sending data to the API.
+ * It console logs errors and success message. We will swap these out for visible validation later.
+ */
+export function createPost() {
+  document.getElementById("uploadForm").addEventListener("submit", function (event) {
+    event.preventDefault();
+
+    const title = document.getElementById("textareaControl").value;
+
+    let mediaObject = {
+      url: document.getElementById("image-url").value,
+      alt: "",
+    };
+
+    let tags = ["snapgo"];
+
+    try {
+      if (!title || title === "") return; // i just added this hoping to prevent the problem in the future.
+
+      console.log("Upload was successful");
+      const response = sendPost(title, tags, mediaObject);
+      console.log("Response", response);
+      setTimeout(() => {
+        updateOldObject(response);
+      }, 2000);
+      document.getElementById("image-url").value = "";
+      document.getElementById("textareaControl").value = "";
+    } catch (error) {
+      console.log(error);
+      throw Error;
```

`catch block`: to suppress an error
`throw`: to create an error

Understand the differences, they have different purposes. Here you're basically catching an error and throwing it again.

**Suggestion:** remove the throw from the catch block and use the try/catch block in a UI/UX code block e.g event listener.